### PR TITLE
refactor transport socket merge logic

### PR DESCRIPTION
**Pull request by ramaraochavali**
No functional change. Just refactors merge transport logic to simplify if conditions and also when the cluster does not have transport socket avoids proto.Merge call.

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/envoyfilter/cluster_patch.go [hunk 1]
```diff
@@ -90,11 +90,15 @@ func mergeTransportSocketCluster(c *cluster.Cluster, cp *model.EnvoyFilterConfig
...
-	} else if cpValueCast.GetTransportSocket() != nil && c.GetTransportSocket() != nil {
-		if cpValueCast.GetTransportSocket().Name == c.GetTransportSocket().Name {
-			tsmPatch = c.GetTransportSocket()
-		} else {
-			// There is a name mismatch, so we cannot do a deep merge. Instead just replace the transport socket
+		// Now check if this matches if existing cluster's transport socket.
+		if c.GetTransportSocket() != nil {
+			if cpValueCast.GetTransportSocket().Name == c.GetTransportSocket().Name {
+				tsmPatch = c.GetTransportSocket()
+			}
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/envoyfilter/cluster_patch.go:97]: I would suggest, although in theory ` c.GetTransportSocket() != nil` and ·`len(c.GetTransportSocketMatches()) > 0` can not be true simultanously 
```
if cpValueCast.GetTransportSocket() != nil {
    if len(c.GetTransportSocketMatches()) > 0 {
    } else if c.GetTransportSocket() != nil{
    }
```

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/envoyfilter/cluster_patch.go]: Ok. Changed.

File: pilot/pkg/networking/core/v1alpha3/envoyfilter/cluster_patch.go [hunk 2]
```diff
@@ -75,26 +75,30 @@ func mergeTransportSocketCluster(c *cluster.Cluster, cp *model.EnvoyFilterConfig
...
+	if cpValueCast.GetTransportSocket() != nil {
+		// First check if the transport socket matches with any cluster transport socket matches.
+		if len(c.GetTransportSocketMatches()) > 0 {
+			for _, tsm := range c.GetTransportSocketMatches() {
+				if tsm.GetTransportSocket() != nil && cpValueCast.GetTransportSocket().Name == tsm.GetTransportSocket().Name {
+					tsmPatch = tsm.GetTransportSocket()
+					break
+				}
+			}
+			if tsmPatch == nil && len(c.GetTransportSocketMatches()) > 0 {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/envoyfilter/cluster_patch.go:88]: `len(c.GetTransportSocketMatches()) > 0` checked in L81

File: pilot/pkg/networking/core/v1alpha3/envoyfilter/cluster_patch.go [hunk 3]
```diff
@@ -75,26 +75,30 @@ func mergeTransportSocketCluster(c *cluster.Cluster, cp *model.EnvoyFilterConfig
...
-			return true, nil
-		}
-	} else if cpValueCast.GetTransportSocket() != nil && c.GetTransportSocket() != nil {
-		if cpValueCast.GetTransportSocket().Name == c.GetTransportSocket().Name {
-			tsmPatch = c.GetTransportSocket()
-		} else {
-			// There is a name mismatch, so we cannot do a deep merge. Instead just replace the transport socket
+		// This means either there is a name mismatch or cluster does not have transport socket matches/transport socket.
+		// We cannot do a deep merge. Instead just replace the transport socket
+		if tsmPatch == nil {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/envoyfilter/cluster_patch.go:101]: Isn't this shoube under `else if c.GetTransportSocket() != nil`

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/envoyfilter/cluster_patch.go]: It really does not matter. But refactored a bit more to simplify further. PTAL.

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**